application.go

```diff
@@ -7,55 +7,55 @@ import (
-	Status string `json:"status"`
+	Name               string         `json:"name" schema:"name"`
+	ID                 string         `json:"id"`
+	NetworkID          string         `json:"network_id" validate:"required" schema:"network_id"`
+	FirewallID         string         `json:"firewall_id" schema:"firewall_id"`
+	Image              *string        `json:"image,omitempty"  schema:"image"`
+	Size               string         `json:"size"  schema:"size"`
+	ProcessInfo        []ProcInfo     `json:"process_info,omitempty"`
+	GitInfo            *GitInfo       `json:"git_info,omitempty" schema:"git_info"`
+	Config             ObservedConfig `json:"config,omitempty"`
+	AppIP              string         `json:"app_ip,omitempty"`
+	Domains            []string       `json:"domains,omitempty"`
+	PublicIPv4Required bool           `json:"public_ipv4_required" schema:"public_ipv4_required"`
+	Status             string         `json:"status"`
 }
 
-// ApplicationConfig describes the parameters for a new CivoApp
-type ApplicationConfig struct {
-	Name        string   `json:"name" validate:"required"`
-	NetworkID   string   `json:"network_id" validate:"required"`
-	Description string   `json:"description"`
-	Size        string   `json:"size"`
-	SSHKeyIDs   []string `json:"ssh_key_ids,omitempty"`
+// GitInfo holds the git information for the application
+type GitInfo struct {
+	GitURL         string          `json:"git_url" schema:"git_url"`
+	GitToken       string          `json:"git_token" schema:"git_token"`
+	PullPreference *PullPreference `json:"pull_preferences,omitempty" schema:"pull_preferences"`
 }
 
-// UpdateApplicationRequest is the struct for the UpdateApplication request
-type UpdateApplicationRequest struct {
-	Name        string        `json:"name"`
-	Advanced    bool          `json:"advanced"`
-	Image       string        `json:"image" `
-	Description string        `json:"description"`
-	ProcessInfo []ProcessInfo `json:"process_info"`
-	Size        string        `json:"size" schema:"size"`
-	SSHKeyIDs   []string      `json:"ssh_key_ids" `
-	Config      []EnvVar      `json:"config"`
-	Domains     []string      `json:"domains"`
+// PullPreference determines which tag/branch should the image be built from.
+// If both are specified, the tag will be used(Tags>Branches)
+// If neither is specified, main/master will be used (main > master)
+type PullPreference struct {
+	Tag    *string `json:"tag,omitempty" schema:"tag"`
+	Branch *string `json:"branch,omitempty" schema:"branch"`
 }
 
-// PaginatedApplications returns a paginated list of Application object
-type PaginatedApplications struct {
-	Page    int           `json:"page"`
-	PerPage int           `json:"per_page"`
-	Pages   int           `json:"pages"`
-	Items   []Application `json:"items"`
+// ObservedConfig defines the observed state of EnvVar
+type ObservedConfig struct {
+	Env          []EnvVar    `json:"env_vars,omitempty"`
+	LastSyncedAt metav1.Time `json:"last_sycned_at,omitempty"`
```

Comment:
Can you see if we send it as a string and accept as a string? We won't have to import this depedency

Reply:
We aren't sending strings. We are trying to send `ObservedConfig` as an array with env vars only, but they aren't working.

Reply:
I meant send metav1.Time as a string